core/io/pread_spec.rb

```diff
@@ -26,11 +26,73 @@
       buffer.should == "567"
     end
 
+    it "shrinks the buffer in case of less bytes read" do
+      buffer = "foo"
+      @file.pread(1, 0, buffer)
+      buffer.should == "1"
+    end
+
+    it "grows the buffer in case of more bytes read" do
+      buffer = "foo"
+      @file.pread(5, 0, buffer)
+      buffer.should == "12345"
+    end
+
     it "does not advance the file pointer" do
       @file.pread(4, 0).should == "1234"
       @file.read.should == "1234567890"
     end
 
+    it "returns an empty string for maxlen = 0, regardless of the offset" do
+      @file.pread(0, 4).should == ""
+      @file.pread(0, 400).should == ""
+    end
+
+    it "does not reset the buffer when reading with maxlen = 0" do
+      buffer = "foo"
+      @file.pread(0, 4, buffer)
+      buffer.should == "foo"
+
+      @file.pread(0, 400, buffer)
+      buffer.should == "foo"
+    end
+
+    it "ignores the current offset" do
+      @file.pos = 3
+      @file.pread(4, 0).should == "1234"
+    end
```

Comment:
**suggestion:** I would put this case a bit closer to the beginning near a case `"does not advance the file pointer"` as far as it is not an edge case but core behaviour of the method.